### ISSUE TITLE
Settings for payment processors are not unique in 4.7

### ISSUE THREAD

**Issue by scardinius** [Project60/org.project60.sepa#414]
CiviCRM 4.7 doesn't use `civicrm_setting.group_name` in order to distinguish settings. This means that `civicrm_setting.name` has to be unique! 

Sepa has settings `pp1`, `pp2`... as a container for creditor_id for selected payment processor. There were also different group_name, 'SEPA Direct Debit PP' and 'SEPA Direct Debit PP Test'.

After upgrade below code returns the same value (based on a greater civicrm_setting.id)

```php
$creditor_id = CRM_Core_BAO_Setting::getItem('SEPA Direct Debit PP', 'pp'.$pp_id);
$test_creditor_id = CRM_Core_BAO_Setting::getItem('SEPA Direct Debit PP Test', 'pp'.$pp_id);
```

I'd like to change name into

* `pp_live1`
* `pp_test1`

**Comment by systopia** [comment]
Thanks @scardinius for reporting. Would you mind creating a PR?

**Comment by scardinius** [comment]
@systopia yes, I'm working on it now :)

**Comment by scardinius** [comment]
@systopia This fix handles getting and setting values but not migrating from old names to new. Do you have good idea? After migrating to 4.7 it's not possible for script/php to decide which value is for live or test payment processor...

**Comment by systopia** [comment]
> Do you have good idea? After migrating to 4.7 it's not possible for script/php to decide which value is for live or test payment processor...

I know, we had the same problem in another extension (de.systopia.donrec). We ended up putting upgrader code into the extension and added the instruction to upgrade the extension (on 4.6) _before_ upgrading CiviCRM (to 4.7)...

**Comment by systopia** [comment]
released in 1.2.4
